packages/flame/lib/src/components/timer_component.dart

```diff
@@ -0,0 +1,36 @@
+import '../../components.dart';
+
+/// Simple component which wraps a [Timer] instance allowing it to be easily
+/// used inside a FlameGame game.
+class TimerComponent extends Component {
+  late final Timer timer;
+  final bool removeOnFinish;
+
+  TimerComponent({
+    required double limit,
+    bool repeat = false,
+    bool autoStart = false,
+    this.removeOnFinish = false,
+  }) {
+    timer = Timer(limit, repeat: repeat, callback: tick);
+
+    if (autoStart) {
+      timer.start();
+    }
+  }
+
+  /// Called everytime the [timer] reached a tick
+  /// default implementaiton is a no-op, override this
+  /// to add custom logic.
+  void tick() {}
```

Comment:
I think this is a bit of a confusing name for the callback.

Reply:
Maybe it is more useful if we can pass in the callback instead of having an implementation in the class, because now you have to extend this component to use it, but many times it is only the callback that you want to change, so maybe it would be easier to have it as an argument like for the timer class, wdyt?

Reply:
The idea of having a method here is so we can have a more "self contained" widget. Just like I refactored on the game in game example. Now all the logic is in the component.

But maybe we can have both things here, I will think on something and update here

Reply:
The default callback argument can be null maybe and if it is specified both are called? :thinking: 
But maybe it is confusing to have two places for the callbacks.
It would be very nice to be able to just do `TimerComponent(limit: 5, callback: () => add(Enemy())` instead of having to create a new component for that.

Reply:
Yeah, that is what I am thinking of, have an argument and have the method. If the callback is specified we call it, otherwise the class method is called.

It is a shame that dart don't allow us to do method tear of on the constructors, because if it did, this would not be necessary at all.

I will implement this later.

Reply:
@spydon  Added the option to pass a callback as well, lmk what you think

Reply:
Looks good! I think the argument and the method should have the same name though, maybe rename `tick` to callback? Or if there is a better name we can give both.